diff --git a/Makefile b/Makefile
index d3874b2..a4c8094 100644
--- a/Makefile
+++ b/Makefile
@@ -548,7 +548,9 @@ CFLAGS_FOR_TARGET=-ffunction-sections -fdata-sections			\
 -Os -fno-unroll-loops
 CXXFLAGS_FOR_TARGET=$(CFLAGS_FOR_TARGET) -fno-exceptions
 
-CFLAGS="-O2 -g -fgnu89-inline"
+CFLAGS="-O2 -g -fgnu89-inline -Wno-error=incompatible-function-pointer-types"
+BREW_PREFIX?=$(shell brew --prefix 2>/dev/null || { uname -m | grep -q '^arm64$$' && printf /opt/homebrew || printf /usr/local; })
+GCC_HOST_DEPS=--with-gmp=$(BREW_PREFIX) --with-mpfr=$(BREW_PREFIX) --with-mpc=$(BREW_PREFIX)
 
 .PHONY: build-gcc
 build-gcc stamps/build-gcc: stamps/install-binutils stamps/prep-gcc
@@ -558,6 +560,7 @@ build-gcc stamps/build-gcc: stamps/install-binutils stamps/prep-gcc
 	popd ; \
 	CFLAGS=$(CFLAGS) ../../gcc-$(GCC_VERSION)/configure --prefix="$(PREFIX)"		\
 	--target=$(TARGET) --enable-languages="c" --with-gnu-ld		\
+	$(GCC_HOST_DEPS)						\
 	--with-gnu-as --with-newlib --disable-nls --disable-libssp	\
 	--with-dwarf2 --enable-version-specific-runtime-libs --disable-libstdcxx-pch	\
 	--disable-shared						\
@@ -600,6 +603,7 @@ build-final-gcc stamps/build-final-gcc: stamps/install-binutils stamps/install-g
 	popd ; \
 	CFLAGS=$(CFLAGS) ../../gcc-$(GCC_VERSION)/configure --prefix=$(PREFIX) \
 	--target=$(TARGET) $(DEPENDENCIES) --enable-languages="c,c++" --with-gnu-ld \
+	$(GCC_HOST_DEPS) \
 	--with-gnu-as --with-newlib --disable-nls --disable-libssp \
 	--with-dwarf2 --enable-version-specific-runtime-libs --disable-libstdcxx-pch \
 	--disable-shared --enable-__cxa_atexit \
